pennylane/ops/qubit/matrix_ops.py

```diff
@@ -261,6 +261,10 @@ class DiagonalQubitUnitary(DiagonalOperation):
     par_domain = "A"
     grad_method = None
 
+    @classmethod
+    def _matrix(cls, *params):
+        return qml.math.diag(cls._eigvals(*params))
```

Comment:
This op was the only one relying on `DiagonalOperation`'s automatic matrix method.

Reply:
😆